src/Evolution/Systems/Cce/AnalyticSolutions/RotatingSchwarzschild.hpp

```diff
@@ -0,0 +1,144 @@
+      "Analytic solution representing a Schwarzschild black hole in a rotating "
+      "frame"};
+
+  WRAPPED_PUPable_decl_template(RotatingSchwarzschild);  // NOLINT
+
+  explicit RotatingSchwarzschild(CkMigrateMessage* /*unused*/) noexcept {}
+
+  // clang doesn't manage to use = default correctly in this case
+  // NOLINTNEXTLINE(modernize-use-equals-default)
+  RotatingSchwarzschild() noexcept {};
+
+  RotatingSchwarzschild(double extraction_radius, double mass,
+                        double frequency) noexcept;
+
+  std::unique_ptr<WorldtubeData> get_clone() const noexcept override;
+
+ protected:
+  /// A no-op as the rotating Schwarzschild solution does not have substantial
+  /// shared computation to prepare before the separate component calculations.
+  void prepare_solution(const size_t /*output_l_max*/,
+                        const double /*time*/) const noexcept override {}
+
+  /*!
+   * \brief Compute the spherical coordinate metric from the closed-form
+   * rotating Schwarzschild metric.
+   *
+   * \details The rotating Schwarzschild takes the coordinate form
+   * \cite Barkett2019uae,
+   *
+   * \f{align}{
+   * ds^2 = -\left(1 - \frac{2 M}{r} - \omega^2 r^2 \sin^2 \theta\right) dt^2
+   * + \frac{1}{1 - \frac{2 M}{r}} dr^2
+   * + 2 \omega r^2 \sin^2 \theta dt d\phi + r^2 d\Omega^2
+   * \f}
+   */
+  void spherical_metric(
+      gsl::not_null<
+          tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+          spherical_metric,
+      size_t l_max, double time) const noexcept override;
+
+  /*!
+   * \brief Compute the radial derivative of the spherical coordinate metric
+   * from the closed-form rotating Schwarzschild metric.
+   *
+   * \details The rotating Schwarzschild takes the coordinate form
+   * \cite Barkett2019uae,
+   *
+   * \f{align}{
+   * \partial_r g_{a b} dx^a dx^b =& -\left(\frac{2 M}{r^2} - 2 \omega^2 r
+   * \sin^2 \theta\right) dt^2 + dr^2 + 4 \omega r \sin^2 \theta dt d\phi
+   * + 2 r d\Omega^2
+   * \f}
+   */
+  void dr_spherical_metric(
+      gsl::not_null<
+          tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+          dr_spherical_metric,
+      size_t l_max, double time) const noexcept override;
+
+  /*!
+   * \brief The time derivative of the spherical coordinate metric in the
+   * rotating Schwarzschild metric vanishes.
+   */
+  void dt_spherical_metric(
+      gsl::not_null<
+          tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+          dt_spherical_metric,
+      size_t l_max, double time) const noexcept override;
+
+  using WorldtubeData::variables_impl;
+
+  using SphericalMetricData::variables_impl;
+
+  /// The News vanishes in the rotating Schwarzschild metric
+  void variables_impl(
+      gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, -2>>*> news,
+      size_t l_max, double time,
+      tmpl::type_<Tags::News> /*meta*/) const noexcept override;
+
+  double frequency_ = 0.0;
+  double mass_ = 1.0;
```

Comment:
In other analytic solutions we set these to `NaN` and instead give the arguments of the constructor default values. Not sure how strongly we follow that convention.